### PR TITLE
[Frontend] Redux Toolkit Setup

### PR DESCRIPTION
**Pull request by brixtuyor**
# Related Links
- http://localhost:8000/signup
- https://app.asana.com/0/1202403928837322/1202486786890552/f

# Description
- Redux Toolkit integration
- Includes part of the [Backend] User task for testing purposes (signup)
- Create a database, in my case I named it `elearning`
- Finally, run `php artisan migrate` to persist the tables

# Notes
- `@reduxjs/toolkit` package is added, please run `npm install`
- Please update your environment variables for database. You may refer to the configuration below:
image

# Screenshots
image
image

### CODE REVIEW

File: resources/js/store/user/api/userApi.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const userApi = createApi({
+  reducerPath: "userApi",
+  baseQuery: fetchBaseQuery({ baseUrl: "http://127.0.0.1:8000/api/" }),
```
Comment (roseaugusto) [inline resources/js/store/user/api/userApi.ts]: you can declare this in .env so that we can efficiently re use the base route from anywhere

**Comment by roseaugusto** [comment]
@brixtuyor
1. **I have confirmation about the terminal errors in my end. Please confirm** 🙇🏼 
image


2. **For `.env variables`, no need to include this since your environment and mine is different, I could be using another port or another db server so not unless we setup infrastructures application (ex. docker, vagrant) no need to have the same values in other user's end.**
image

**Comment by brixtuyor** [comment]
> @brixtuyor
> 
> 1. **I have confirmation about the terminal errors in my end. Please confirm** 🙇🏼
> 

Is this after you ran `npm install`?

**Comment by roseaugusto** [comment]
> Is this after you ran `npm install`?

@brixtuyor offt sorry my bad, it's all goods now.

File: resources/js/store/user/api/userApi.ts (hunk 1 above)
Comment (brixtuyor) [reply resources/js/store/user/api/userApi.ts]: Done, moved to `env` file

**Comment by brixtuyor** [comment]
> > Is this after you ran `npm install`?
> 
> @brixtuyor offt sorry my bad, it's all goods now.

It's okay no worries

File: resources/js/store/user/api/userApi.ts (hunk 1 above)
Comment (brixtuyor) [reply resources/js/store/user/api/userApi.ts]: image

File: resources/js/store/user/api/userApi.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+const fetchBaseUrl = process.env.MIX_FETCH_BASE_URL;
```
Comment (roseaugusto) [inline resources/js/store/user/api/userApi.ts]: For the naming, is it only for fetching? or you will also use this for other HTTP methods?
```suggestion
const fetchBaseUrl = process.env.MIX_BASE_URL;
```

Comment (brixtuyor) [reply resources/js/store/user/api/userApi.ts]: @roseaugusto You actually have a point, thank you will change it now.

**Review by roseaugusto — APPROVED** [review]
Approved!

**Pull request merged**